cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -89,6 +90,77 @@ public GradebookDTO getGradebook(@PathVariable("id") Integer assignmentId  ) {
 		return gradebook;
 	}
 	
+	@PostMapping("/course/{course_id}/add")
+	public void addAssignment(@RequestBody AssignmentDTO assignmentDTO, @PathVariable int course_id) {
+		
+		// check that this request is from the course instructor 
+		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
+		
+		// grab requested course and validate permissions if it exists
+		Course c = courseRepository.findById(course_id).orElse(null);
+		
+		if (!c.getInstructor().equals(email)) {
```

Comment:
If c is null from line 100 you might get an error here by calling a method (the .getInstructor()) on a null...similar thought for lines 124 and 148

Reply:
changes implemented